eng/ci/consolidate-artifacts.yml

```diff
@@ -0,0 +1,304 @@
+      incrementalSDLSourceAnalysis: true
+    pool:
+      name: 1es-pool-azfunc
+      image: 1es-windows-2022
+      os: windows
+    sdl:
+      codeql:
+         compiled:
+           enabled: true
+         runSourceLanguagesInSourceAnalysis: true
+     
+    stages:
+    - stage: ConsolidateArticacts
+      displayName: "Assemble Artifacts"
+      jobs:
+      - job: ProcessNuGetPackage
+        displayName: "Process NuGet Package"
+        templateContext:
+          outputs:
+            - output: pipelineArtifact
+              displayName: Publish nuget package
+              targetPath: $(Pipeline.Workspace)/staging
+              artifact: drop-nuget-package
+        steps:
+        - checkout: none
+
+        - download: core-tools-default
+          artifact: drop
+          displayName: "Download out-of-proc artifact"
+
+        - powershell: |
+            $currentRunName = $env:BUILD_REASON
+            Write-Host "Current pipeline run name: $currentRunName"
+            # Retrieve environment variables
+            $defaultArtifactAlias = $env:DEFAULT_ARTIFACT_ALIAS
+            $defaultArtifactName = $env:DEFAULT_ARTIFACT_NAME
+
+            # Get the current working directory
+            $currentDirectory = "$(Pipeline.Workspace)"
+
+            # Construct the path using current directory, and environment variables
+            $artifactPath = Join-Path -Path $currentDirectory -ChildPath "$defaultArtifactAlias\$defaultArtifactName"
+
+            # Define the regex pattern to match the nupkg file (digit.digit.4digits)
+            $regexPattern = "^Microsoft\.Azure\.Functions\.CoreTools\.\d+\.\d+\.\d{4}\.nupkg$"
+
+            # Look for the first nupkg file that matches the pattern in the constructed path
+            $fileToMove = Get-ChildItem -Path $artifactPath -Filter "*.nupkg" | Where-Object { $_.Name -match $regexPattern } | Select-Object -First 1
+
+            # Check if a matching file was found
+            if ($fileToMove -eq $null) {
+                Write-Host "No .nupkg file matching the pattern was found in $artifactPath"
+                exit 1
+            }
+
+            # Define the destination path in $(Pipeline.Workspace)/staging
+            $nugetPackageDirectory = Join-Path -Path $(Pipeline.Workspace) -ChildPath "staging"
+
+            # Create the staging directory if it doesn't exist
+            if (-not (Test-Path $nugetPackageDirectory)) {
+                New-Item -Path $nugetPackageDirectory -ItemType Directory
+                Write-Host "Directory created at: $nugetPackageDirectory"
+            }
+
+            # Move the file to the staging directory
+            Move-Item -Path $fileToMove.FullName -Destination $nugetPackageDirectory -Force
+
+            Write-Host "File $($fileToMove.Name) moved to $nugetPackageDirectory"
+          displayName: 'Process Microsoft.Azure.Functions.CoreTools NuGet Package'
+
+      - job: AssembleArtifacts
+        strategy:
+          matrix:
```

Comment:
Could this be simplied with a template where you pass in an array of the platforms/arch you want to build?